jobsdb/jobsdb.go

```diff
@@ -2458,6 +2420,12 @@ func (jd *HandleT) markClearEmptyResult(ds dataSetT, workspace string, stateFilt
 		jd.invalidCacheKeyStat.Increment()
 	}
 
+	for _, parameterFilter := range parameterFilters {
+		if !misc.ContainsString(ParameterFilters, parameterFilter.Name) {
+			return
+		}
+	}
```

Comment:
```suggestion
	// Skip the cache if a parameter filter is provided that does not belong to the caching key
	for _, parameterFilter := range parameterFilters {
		if !misc.ContainsString(ParameterFilters, parameterFilter.Name) {
			return
		}
	}
```

Reply:
Isn't it more appropriate to test whether all parameter filters are provided or not?

E.g. if we were using 2 parameter filters for the caching key and the user provided only 1 of them as a parameter filter in the query, the cache should be skipped.

This test is now failing

```
t.Run("Test cache with two less parameter filters (destination_id & source_id)", func(t *testing.T) {
	previousParameterFilters := ParameterFilters
	ParameterFilters = []string{"destination_id", "source_id"}
	defer func() {
		ParameterFilters = previousParameterFilters
	}()
	jobsDB := NewForReadWrite("two_params_cache_query_less")
	require.NoError(t, jobsDB.Start())
	defer jobsDB.TearDown()

	destinationID := "destinationID"

	res, err := jobsDB.getUnprocessed(context.Background(), GetQueryParamsT{CustomValFilters: []string{customVal}, ParameterFilters: []ParameterFilterT{{Name: "destination_id", Value: destinationID}}, JobsLimit: 100})
	require.NoError(t, err)
	require.Equal(t, 0, len(res.Jobs), "jobsDB should report 0 unprocessed jobs when using destination_id as filter")

	require.NoError(t, jobsDB.Store(context.Background(), generateJobs(2, destinationID)))
	res, err = jobsDB.getUnprocessed(context.Background(), GetQueryParamsT{CustomValFilters: []string{customVal}, ParameterFilters: []ParameterFilterT{{Name: "destination_id", Value: destinationID}}, JobsLimit: 100})
	require.NoError(t, err)
	require.Equal(t, 2, len(res.Jobs), "jobsDB should report 2 unprocessed jobs when using destination_id as filter, after we added 2 jobs")
})
```